modules/internal/ChapelLocale_forDocs.chpl

```diff
@@ -83,18 +83,31 @@ module ChapelLocale {
     pragma "no doc"
     const parent : locale;
 
-    // To be removed from the required interface once legacy code is adjusted.
-    // Modified in RootLocale.init().
+    pragma "no doc" var nPUsPhysAcc: int;    // HW cores, OS accessible
+    pragma "no doc" var nPUsPhysAll: int;    // HW cores, all
+    pragma "no doc" var nPUsLogAcc: int;     // HW threads, OS accessible
+    pragma "no doc" var nPUsLogAll: int;     // HW threads, all
 
     /*
-      This is the number of processors on this locale.  The term
-      *cores* here is a misnomer.  The value is actually the number
-      of instances of the CPU architecture, so for a locale with a
-      multithreaded hardware architecture, it's the number of
-      threads.  Also, only processors actually available to the
-      program are in this count.  If the OS has made some of them
-      off-limits to the program, for example, those are not
-      included.
+      A *processing unit* or *PU* is an instance of the processor
+      architecture, basically the thing that executes instructions.
+      ``numPUs()`` tells how many of these are present on this locale.
+      If ``physical==true`` (the default) this is the number of physical
+      PUs, commonly known as *cores*.  Otherwise it is the number of
+      hardware threads (such as hyperthreads), which on a multithreaded
+      hardware architecture would typically be more numerous.  If
+      ``accessible==true`` (the default) then the count reflects only
+      those PUs the program has access to, due to limiting by the OS for
+      example.  Otherwise, all PUs that seem to exist are included.
+     */
+    inline
+    proc numPUs(physical: bool = true, accessible: bool = true)
+      return if physical
+             then if accessible then nPUsPhysAcc else nPUsPhysAll
+             else if accessible then nPUsLogAcc else nPUsLogAll;
+
+    /*
+      ``numCores`` is deprecated and will be removed in Chapel 1.14.
```

Comment:
This file is new to me (I think it showed up while I was traveling) -- what would it take to retire ChapelLocale_forDocs.chpl to avoid having to duplicate code/documentation like this?